src/uploader.c

```diff
@@ -2451,36 +2475,50 @@ static void prepare_upload_state(uv_work_t *work)
 
     state->encrypted_file_name = encrypted_file_name;
 
-    // Calculate deterministic file id from encrypted file name
-    char *file_id = calloc(FILE_ID_SIZE + 1, sizeof(char));
-    if (!file_id) {
+    // Get random index used for encryption
+    uint8_t *index = calloc(SHA256_DIGEST_SIZE + 1, sizeof(uint8_t));
```

Comment:
==27331== 33 bytes in 1 blocks are definitely lost in loss record 28 of 57
==27331==    at 0x4C2FB55: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==27331==    by 0x4156B7: prepare_upload_state (uploader.c:2479)
==27331==    by 0x54EF2DE: ??? (in /usr/lib/x86_64-linux-gnu/libuv.so.1.0.0)
==27331==    by 0x54FCCB6: ??? (in /usr/lib/x86_64-linux-gnu/libuv.so.1.0.0)
==27331==    by 0x5A1B6B9: start_thread (pthread_create.c:333)
==27331==    by 0x5D3782C: clone (clone.S:109)
==27331==